### ISSUE TITLE
examples simple-rag bug: split_paragraphs isn't working correctly

### ISSUE THREAD

**Issue by Kael-DWT** [livekit/agents#870]
livekit.agents.tokenize._basic_paragraph.split_paragraphs 

```python
def split_paragraphs(text: str) -> list[tuple[str, int, int]]:
    """
    Split the text into paragraphs.
    Returns a list of paragraphs with their start and end indices of the original text.
    """
    matches = re.finditer(r"\n{2,}", text)
    paragraphs = []

    for match in matches:
        paragraph = match.group(0)
        start_pos = match.start()
        end_pos = match.end()
        paragraphs.append((paragraph.strip(), start_pos, end_pos))

    return paragraphs
```

Is this regex written incorrectly? It should be like this.
```python
matches = re.finditer(r".+\n{2,}", text)
```

**Comment by LIHUA919** [comment]
I also encountered the same problem

**Comment by m-tabish** [comment]
Kindly assign me this under Hacktober 2024

**Comment by m-tabish** [comment]
Kindly assign this to me, I would like to work on this.

**Comment by davidzhao** [comment]
@m-tabish of course, please go ahead and submit a PR. it'd be great if you are able to add test coverage here as well, so we won't break it unintentionally again.

**Comment by m-tabish** [comment]
Aren't you guys in Hacktoberfest? Is you are kindly add a tag here

**Comment by davidzhao** [comment]
fixed in #896
